src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -76,7 +77,44 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
 
                         string challengeData = challenge.ChallengeData;
 
-                        string spn = "HTTP/" + authUri.IdnHost;
+                        // Calculate SPN (Service Principal Name) using the host name of the request.
+                        // Use the request's 'Host' header if available. Otherwise, use the request uri.
+                        // Ignore the 'Host' header if this is proxy authentication since we need to use
+                        // the host name of the proxy itself for SPN calculation.
+                        string hostName;
+                        if (!isProxyAuth && request.HasHeaders && request.Headers.Host != null)
+                        {
+                            // Use the host name without any normalization.
+                            hostName = request.Headers.Host;
+                            if (NetEventSource.IsEnabled)
+                            {
+                                NetEventSource.Info(connection, $"Authentication: {challenge.AuthenticationType}, Host: {hostName}");
+                            }
```

Comment:
Nit: we've generally treated the "if (NetEventSource.IsEnabled) " as a prefix to calls to NetEventSource, necessary only for perf, and have thus generally done all of it on one line.

Reply:
I copied the code from the master branch. And there are several places already in this file that use this multi-line pattern:

https://github.com/dotnet/corefx/blob/7350b23f8f4f6bd01e935615e463e87500a00b05/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs#L48-L51

https://github.com/dotnet/corefx/blob/7350b23f8f4f6bd01e935615e463e87500a00b05/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs#L79-L82

https://github.com/dotnet/corefx/blob/7350b23f8f4f6bd01e935615e463e87500a00b05/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs#L117-L120

etc.

So, I didn't want to make it look different from everything else in this file. :)